src/components/src/ImageContainer/image.container.tsx

```diff
@@ -19,69 +19,81 @@
+  alt: string,
+  /** The onLoadData */
+  onLoadData?: any,
 }
 
 function ImageContainer(props: ImageContainerProps) {
   const {
-    fileName, imgUrl, className, isSkuImage, onLoadData,
+    imgUrl, pictureClassName, imgClassName, isSkuImage, types, sizes, fileName, alt, onLoadData,
   } = props;
 
-  const imageSource = isSkuImage ? Config.skuImagesUrl.replace('%sku%', fileName) : Config.siteImagesUrl.replace('%fileName%', fileName);
+  const imageSizes = sizes === undefined ? Config.ImageContainerSrcs.sizes : sizes;
+  const imageTypes = types === undefined ? Config.ImageContainerSrcs.types : types;
+  let imgPrefix = '';
+  const imgAlt = alt != null ? alt : '';
+
+  if (isSkuImage) {
+    imgPrefix = Config.skuImagesUrl;
+  } else {
+    imgPrefix = Config.siteImagesUrl;
+  }
+
+  const [error, setError] = useState(false);
 
   const handleError = (e, defaultImgUrl) => {
-    const { src } = e.target;
-    if (Config.imageFileTypes && Config.imageFileTypes.enable && e.target['data-source'] !== 'default') {
-      const fallbackTypes = Config.imageFileTypes.types;
-      const initType = imgUrl.replace(/.*(?=\.)/g, '');
-      const types = (fallbackTypes && fallbackTypes.length > 0) ? [
-        initType,
-        ...fallbackTypes.filter(fileType => fileType !== initType),
-      ] : [];
-      const currentType = src.replace(/.*(?=\.)/g, '');
-      const i = types.indexOf(currentType);
-      if (types[i + 1]) {
-        e.target.src = src.replace(currentType, types[i + 1]);
-      } else {
-        e.target.src = isSkuImage ? imgPlaceholder : defaultImgUrl;
-        e.target['data-source'] = 'default';
-      }
-    } else if (e.target['data-source'] !== 'default') {
-      e.target.src = isSkuImage ? imgPlaceholder : defaultImgUrl;
-      e.target['data-source'] = 'default';
-    }
+    e.currentTarget.src = isSkuImage ? imgPlaceholder : defaultImgUrl;
+    setError(true);
   };
 
-  return (
-    <img
-      className={className}
-      alt=""
-      src={imageSource}
-      onError={e => handleError(e, imgUrl)}
-      onLoad={onLoadData}
-    />
-  );
+  const generateSrcSet = type => imageSizes.reduce((acc, imageSize) => {
+    if (acc === '') {
+      return `${acc}${imgPrefix.replace('%fileName%', `${type}/${fileName}-${imageSize}w.${type} ${imageSize}w`)}`;
+    }
+    return `${acc}, ${imgPrefix.replace('%fileName%', `${type}/${fileName}-${imageSize}w.${type} ${imageSize}w`)}`;
+  }, '');
+
+  if (!error) {
+    return (
+      <picture className={pictureClassName} key={fileName}>
+        {imageTypes.map(type => <source onError={e => handleError(e, imgUrl)} key={`fileName${type}`} srcSet={generateSrcSet(type)} type={`image/${type}`} />)}
+        <img className={imgClassName} alt="" src={imgUrl} key={fileName} onLoad={onLoadData} onError={e => handleError(e, imgUrl)} />
+      </picture>
+    );
+  }
+
+  return (<img className={imgClassName} alt="" src={imgUrl} onLoad={onLoadData} onError={e => handleError(e, imgUrl)} />);
```

Comment:
No point in handling error if error already happened. This might create an endless loop.